client/src/QAComponents/AnswerList.jsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom'
+import ReactDOM from 'react-dom'
+import axios from 'axios'
+import NewAnswer from './NewAnswer.jsx'
+
+const AnswerList = (props) => {
+  const [state, setState] = useState()
+  const [answers, setAnswers] = useState(['test1, test2'])
+  const [moreAnswers, setMoreAnswers] = useState(['dummy', 'data'])
+  const [moreQuestions, setMoreQuestions] = useState(['dummy', 'data'])
+  const [didMount, setDidMount] = useState(false)
+  const [helpfulAns, setHelpfulAns] = useState(0)
+  let renderMoreAnswers = <div />
+  // const [questionId, setQuestionId] = useState(props.question_id);
+
+  // SERVER REQUESTS
+  const getAllAnswers = (question, callback) => {
+    if (question === undefined) {
+      return
+    }
+    axios.get('/qa/answers?question_id=' + question)
+      .then(data => {
+        callback(null, data)
+      })
+      .catch(err => console.log(err))
+  }
+
+  const reportAnswer = (e) => {
+    const answerId = e.target.parentNode.parentNode.parentNode.id
+    e.preventDefault()
+    axios.put('/qa/answers/report/?answer_id=' + answerId)
+      .then(data => {
+        console.log(data)
+      })
+      .catch(err => {
+        console.error(err)
+      })
+  }
+
+  const helpfulAnswer = (e) => {
+    const answerId = e.target.parentNode.parentNode.parentNode.id
+    console.log('clicked')
+    e.preventDefault()
+    axios.put('/qa/answers/helpful/?answer_id=' + answerId)
+      .then(data => {
+        console.log(helpfulAns)
+        return setHelpfulAns(!helpfulAns)
+      })
+      .catch(err => {
+        console.error(err)
+      })
+  }
+
+  // INITIAL RENDER
+  const initialize = (questionId, callback) => {
```

Comment:
I would recommend moving away from a callback pattern here. I would propose either returning a promise from you're own functions or maybe moving to async/await if possible.